docs/6-rag/5-rag-pipelines.md

```diff
@@ -0,0 +1,120 @@
+- Initialize LlamaStack client and model configuration
+- Configure embedding models optimized for academic content
+- Set up document processing parameters (timeouts, chunk sizes)
+
+**Stage 2: Docling Document Processing** 🧠  
+- Transform complex PDFs using Docling's advanced analysis
+- Preserve tables, formulas, figures, and document structure
+- Handle multi-column layouts typical of academic papers
+
+**Stage 3: Vector Database Creation** 🗄️
+- Create Milvus database optimized for document intelligence
+- Configure semantic search for educational content
+- Set up enhanced metadata storage
+
+**Stage 4: Document Ingestion** 📊
+- Store intelligently-processed content with rich metadata
+- Create semantic embeddings preserving document structure
+- Enable advanced querying capabilities
+
+**Stage 5: RAG Testing & Validation** ❓
+- Execute test queries demonstrating document intelligence
+- Validate system performance with academic content
+- Generate quality metrics and capability reports
+
+## 🎯 Hands-On: Running Your Production Pipeline
+
+Time to deploy your document intelligence RAG system in production!
+
+1. **Create the Pipeline Storage Volume**
+
+   First, set up persistent storage for content transfer between pipeline stages:
+
+   ```bash
+   oc apply -f canopy/6-rag/4-kfp-pipeline-pvc.yaml -n -n <USER_NAME>-prod
+   ```
+
+   This creates a `canopy-workspace-pvc` that allows pipeline components to share processed document content efficiently.
+
+2. **Configure Your Educational Pipeline**
+
+   Open `canopy/6-rag/4-kfp_pipeline.py` and review the configuration optimized for academic content:
+
+   ```python
+   arguments = {
+       "document_url": "https://arxiv.org/pdf/2404.14661",        # Research paper URL
+       "test_queries": [
+           "What is the PRFXception mentioned in the document?",   # Academic concept query
+       ],
+       "embedding_model": "all-MiniLM-L6-v2",                     # Optimized for education
+       "embedding_dimension": 384,                                 # Performance-balanced
+       "chunk_size_tokens": 512,                                   # Academic content chunks
+       "vector_provider": "milvus",                                # Production vector DB
+       "docling_service": "http://docling-v0-7-0-predictor.ai501.svc.cluster.local:5001",
+       "processing_timeout": 180,                                  # 3 min for complex docs
+       "llama_stack_url": "http://llama-stack-service:8321",      # AI inference service
+       "model_id": "llama32",                                      # Educational LLM
+       "temperature": 0.0,                                         # Deterministic responses
+       "max_tokens": 4096                                          # Comprehensive answers
+   }
+   ```
+
+3. **Execute the Production Pipeline**
+
+   Run your document intelligence pipeline by running on `python canopy/6-rag/4-kfp_pipeline.py`
```

Comment:
```suggestion
   Run your document intelligence pipeline by running on `python canopy/6-rag/4-kfp_pipeline.py` in your terminal.
```

Reply:
it's correct :) currently we use the jupyter notebooks but we will update it to run it within the code server as @RHRolun suggested. So the kfp pipeline needs to be updated to be compiled and executed from the terminal instead of the jupyter notebook.